### PR TITLE
SAA-68 now running the integration tests as part of the CI build pipeline.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Include the integration tests as part of the Cirlce CI build pipeline.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -33,6 +33,27 @@ jobs:
...
+      name: hmpps/java
+      tag: "17.0"
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - gradle-{{ checksum "build.gradle.kts" }}
+            - gradle-
+      - run:
+          command: ./gradlew integrationTest
```
Comment (matt-crowson-moj) [inline .circleci/config.yml:47]: Note.  I have removed the `--illegal-access=permit` as this is no longer supported in JDK 17 or above.

This provides a bit more details at [infoq](https://www.infoq.com/news/2021/06/internals-encapsulated-jdk17/).

Comment (timharrison-moj) [reply .circleci/config.yml:47]: Should we be running at v18? If available...

Comment (timharrison-moj) [reply .circleci/config.yml:47]: Same for the `validate` target too? If a v18 executor is available?

**Review by timharrison-moj — APPROVED** [review]
LGTM. Just questions on the executor version at v17?

File: .circleci/config.yml (hunk 1 above)
Comment (matt-crowson-moj) [reply .circleci/config.yml:47]: Ah so we are running JVM 18?  I missed that.  CVL is running at 16.  I'll merge this and try 18.

Comment (timharrison-moj) [reply .circleci/config.yml:47]: Yes - `Dockerfile` uses a Java 18 base image
```
eclipse-temurin:18-jre-jammy
```

 .. and `build.gradle.kts` set to version 18 too 

```
java {
  toolchain.languageVersion.set(JavaLanguageVersion.of(18))
}
```

**Pull request merged**